src/if.js

```diff
@@ -20,7 +20,10 @@ export class If extends IfCore {
   */
   bind(bindingContext, overrideContext) {
     super.bind(bindingContext, overrideContext);
-    this.conditionChanged(this.condition);
+    // Else attribute will trigger change when it's bound
+    if (!this.else || this.else.isBound) {
```

Comment:
Let me recap the problem: 
When `if` is bound, if `condition` is `false` then `else._show()` is called but at this point `else.bind()` hasn't happen so everything crash and burns.

Re-thinking what is supposed to happen during bind, can I suggest a more straightforward solution:
```js
// inside If::bind()
if (this.condition) this._show();

// inside Else::bind()
if (!this.if.condition) this._show();
```

Would that work?
It seems simpler to understand.

Reply:
@jods4 I'm not quite sure if this is possible since the else.bind function won't be called when If's condition has changed. I guess this is why you implemented it that way^^

Reply:
> the else.bind function won't be called when If's condition has changed.

It is called when the component is first bound. It is not meant to be called when the condition later changes.

`bind()` functions are called once at startup for both `if` and `else`, they ensure the template is visible when it is first bound.

If the condition later changes, `If::conditionChanged()` is called and updates both element properly.

Am I missing something?

Reply:
But in this case only the initial render would work or?
If the if condition get's bound to new values the bind function should be evaluated for the if only.

Actually I'm not really sure here if your way would work in all scenarios but this one does and it has not really an impact to change it or I'm wrong here? :)

Reply:
If what I proposed above works (I think it should), it's much simpler and easier to understand so I'd like to include it.
Simpler code is very important for people who will maintain or contribute to Aurelia in the future -- or just because it means that the risk of bug is lower.

In fact I had to read your code several time to figure out how everything clicks together. Conditions like `this.elseVm.isBound` in `If::bind()` really make no sense at first glance -- heck thinking about it again: why call `conditionChanged` in `If::bind()` when `else` was bound, as `else` would actually have called `if.conditionChanged` in its own bind?!

RE your questions: when `condition` changes, `bind` is not called again. Changing the `bind()` implementation doesn't impact that, it only matters when the controller is first connected by Aurelia.